### DISCUSSION TITLE
Source tree structure

### DISCUSSION THREAD

**Discussion by Bill-Kunj** [rchain-community/rgov#188] [General] [answered]
@w2vy @jimscarver @dckc 
The current source tree has all the rholang class files at the top level.
This is like fingernails on a chalkboard to me. I always think of the top-level directory having (mostly) subdirectories that organize everything.
For that matter, deploy-all currently hardcodes the names of the rholang files it deploys. Things could be genericized with a more structured source tree. Then localhost dev environment need not be specific to RGOV.

Let's discuss.

**Comment by dckc** [comment]
I agree that the number of files at the top level is uncomfortable. But `git blame` doesn't work very well across renames/moves (especially via github), and I rely heavily on the history. So it's a tricky balance.

**Reply by Bill-Kunj** [reply]
@dckc @jimscarver @David405 @w2vy I think we should consider creating a ```rholang``` or ```RGOV-src``` directory *somewhere*. Perhaps under ```src```, but that might affect the webpack deployment and -also- be confused with the rholang actions.

**Reply by dckc** [reply]
I have considered it... a few times now. On balance, I don't think moving things around is cost-effective at this point. Perhaps it's a lesson for future projects.

**Comment by Bill-Kunj** [comment]
I believe this discussion is complete now that the rholang directory exists.
@dckc @jimscarver @David405 Someone please confirm by approving an answer so we can close this down.

**Comment by dckc** [comment, marked as answer]
I guess this discussion moved on without me. I'm mildly disappointed that the [annotated version of rholang/core/CrowdFund.rho](https://github.com/rchain-community/rgov/blame/master/rholang/core/CrowdFund.rho) no longer shows my contribution at all. Sigh.

**Reply by Bill-Kunj** [reply]
Staff meetings.... Such a bother. :)
Add your name inside CrowdFund.rho in a PR if you feel it is required.
